lib/rubygems/commands/uninstall_command.rb

```diff
@@ -148,21 +148,30 @@ def uninstall_all
 
   def uninstall_specific
     deplist = Gem::DependencyList.new
+    original_gem_version = {}
+    gems_to_uninstall = {}
```

Comment:
This hash is only used in the second loop, so I would move it to right before it starts to make it more clear.